azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerapps/containerapp/ContainerAppDraft.java

```diff
@@ -414,15 +433,76 @@ private static Secret getSecret(final ImageConfig config) {
     private static RegistryCredentials getRegistryCredential(final ImageConfig config) {
         final ContainerRegistry registry = config.getContainerRegistry();
         if (Objects.nonNull(registry)) {
-            final String username = registry.getUserName();
-            final String password = Optional.ofNullable(registry.getPrimaryCredential()).orElseGet(registry::getSecondaryCredential);
-            final String passwordKey = Objects.equals(password, registry.getPrimaryCredential()) ? "password" : "password2";
-            final String passwordName = String.format("%s-%s", registry.getName().toLowerCase(), passwordKey);
-            return new RegistryCredentials().withServer(registry.getLoginServerUrl()).withUsername(username).withPasswordSecretRef(passwordName);
+            if (StringUtils.isEmpty(config.registryIdentity)) {
+                final String username = registry.getUserName();
+                final String password = Optional.ofNullable(registry.getPrimaryCredential()).orElseGet(registry::getSecondaryCredential);
+                final String passwordKey = Objects.equals(password, registry.getPrimaryCredential()) ? "password" : "password2";
+                final String passwordName = String.format("%s-%s", registry.getName().toLowerCase(), passwordKey);
+                return new RegistryCredentials().withServer(registry.getLoginServerUrl()).withUsername(username).withPasswordSecretRef(passwordName);
+            } else if (StringUtils.equalsIgnoreCase(config.registryIdentity, "system")) {
+                return new RegistryCredentials().withServer(registry.getLoginServerUrl()).withIdentity("system");
+            } else {
+                return new RegistryCredentials().withServer(registry.getLoginServerUrl()).withIdentity(config.registryIdentity);
+            }
         }
         return null;
     }
 
+    @Nullable
+    private ManagedServiceIdentity getManagedServiceIdentity(ImageConfig imageConfig) {
```

Comment:
This is not a pure get method. It contains permission granting. Suggest to clarify in the method name.

Reply:
And it seems only user assigned identity will be returned, which should also clarify in the method name.